### PR TITLE
Extvlcopt support

### PR DESCRIPTION
**Pull request by phunkyfish**
v3.7.0
- Support EXTVCOPT in m3u
- Build helper script for OSX
- Add appveyor.yml

### CODE REVIEW

**Comment by phunkyfish** [comment]
@joaoppp can you check this PR by testing this branch: https://github.com/phunkyfish/pvr.iptvsimple/tree/EXTVLCOPT-support

**Comment by phunkyfish** [comment]
This is failing due to:

```
Scanning dependencies of target p8-platform
[  2%] Creating directories for 'p8-platform'
[  4%] Performing download step (git clone) for 'p8-platform'
Cloning into 'p8-platform'...
fatal: reference is not a tree: 32190045c7eb6883c0662db2f91b4ceeab904fc2
CMake Error at /home/travis/build/kodi-pvr/pvr.iptvsimple/build/build/p8-platform/tmp/p8-platform-gitclone.cmake:75 (message):
  Failed to checkout tag: '32190045c7eb6883c0662db2f91b4ceeab904fc2'
```

Would this be affected by https://github.com/xbmc/xbmc/pull/16417?

**Comment by phunkyfish** [comment]
Ok, this is why: https://github.com/xbmc/platform/pull/5

**Comment by phunkyfish** [comment]
I guess there is no point in approving/merging this until addon builds are fixed @ksooo?

**Comment by phunkyfish** [comment]
@ksooo, ok to go in?

Backport of #244

**Comment by phunkyfish** [comment]
Ok to go in?

File: build-install-mac.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+XBMC_BUILD_ADDON_INSTALL_DIR=$(cd $1/addons/$ADDON_NAME 2> /dev/null && pwd -P)
+
+for DIR in "$BINARY_ADDONS_TARGET_DIR/"macosx*; do
+    if [ -d "${DIR}" ]; then
+	MACOSX_BINARY_ADDONS_TARGET_DIR="${DIR}"
+    fi
+done
+
+if [ -z "$MACOSX_BINARY_ADDONS_TARGET_DIR" ]; then
+  echo "Error: Could not find binary addons build directory at: $BINARY_ADDONS_DIR/macosx*" >&2
```
Comment (AlwinEsch) [inline build-install-mac.sh]: Do you mean here `BINARY_ADDONS_TARGET_DIR`, if yes a small typo?

File: build-install-mac.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  exit 1
+fi
+
+if [ ! -d "$KODI_ADDONS_DIR" ]; then
+  echo "Error: Kodi addons dir does not exist at: $KODI_ADDONS_DIR" >&2
+  exit 1
+fi
+
+cd "$MACOSX_BINARY_ADDONS_TARGET_DIR"
+make
```
Comment (AlwinEsch) [inline build-install-mac.sh:43]: The script try to build all available adding there.
For this place is more: `make ADDONS=$ADDON_NAME`

Comment (AlwinEsch) [reply build-install-mac.sh:43]: The main problem with this is the Kodi's build system which expects addon under `./Kodi/tools/depends/target/binary-addons`.

Without comes:
```
-- ---- Configuring addon /Users/alwin/Development/kodi/tools/depends/target/binary-addons/addons/pvr.iptvsimple/pvr.iptvsimple.txt ----
CMake Error at CMakeLists.txt:429 (message):
  pvr.iptvsimple: invalid or missing addon source directory at
  /Users/alwin/Development/kodi/tools/depends/target/binary-addons/pvr.iptvsimple
```

File: build-install-mac.sh [hunk 3]
```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
```
Comment (AlwinEsch) [inline build-install-mac.sh:2]: Add a `set -e` to exit if some call makes a error.

File: build-install-mac.sh [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+MACOSX_BINARY_ADDONS_TARGET_DIR=""
+KODI_ADDONS_DIR="$HOME/Library/Application Support/Kodi/addons"
+ADDON_NAME=`basename -s .git \`git config --get remote.origin.url\``
+
+if [ ! -d "$BINARY_ADDONS_TARGET_DIR" ]; then
+  echo "Error: Could not find binary addons directory at: $BINARY_ADDONS_TARGET_DIR" >&2
+  exit 1
+fi
+
+XBMC_BUILD_ADDON_INSTALL_DIR=$(cd $1/addons/$ADDON_NAME 2> /dev/null && pwd -P)
```
Comment (AlwinEsch) [inline build-install-mac.sh]: On first start or build before has failed, this not present and stays empty. The `cp`on end  brings then `cp: : No such file or directory`. 
Better to have then with them after `make`.

File: README.md [hunk 5]
```diff
@@ -10,11 +10,37 @@ IPTV Live TV and Radio PVR client addon for [Kodi](http://kodi.tv)
...
 
+### Mac OSX
+
+In order to build the addon on mac the steps are different to Linux and Windows as the cmake command above will not produce an addon that will run in kodi. Instead using make directly as per the supported build steps for kodi on mac we can build the tools and just the addon on it's own. Following this we copy the addon into kodi. Note that we checkout kodi to a separate directory as this repo will only only be used to build the addon and nothing else.
+
+#### Build tools and initial addon build
+
+1. Get the repos
+ * `cd $HOME`
+ * `git clone --branch Leia https://github.com/xbmc/xbmc xbmc-addon`
```
Comment (AlwinEsch) [inline README.md:26]: Maybe better to hold dir name equal with other guides where this is used as `kodi`

File: README.md [hunk 6]
```diff
@@ -10,11 +10,37 @@ IPTV Live TV and Radio PVR client addon for [Kodi](http://kodi.tv)
 ### Linux
 
 1. `git clone --branch Leia https://github.com/xbmc/xbmc.git`
-2. `git clone https://github.com/kodi-pvr/pvr.iptvsimple.git`
+2. `git clone --branch Leia https://github.com/kodi-pvr/pvr.iptvsimple.git`
 3. `cd pvr.iptvsimple && mkdir build && cd build`
 4. `cmake -DADDONS_TO_BUILD=pvr.iptvsimple -DADDON_SRC_PREFIX=../.. -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=../../xbmc/addons -DPACKAGE_ZIP=1 ../../xbmc/cmake/addons`
```
Comment (AlwinEsch) [inline README.md:15]: Just a note 😄. Add-on can also build by a modified way like this.
See https://github.com/AlwinEsch/web.browser.chromium/blob/master/docs/README.macOS.md
But to have a script makes it easier.

File: build-install-mac.sh [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+if [ ! -d "$KODI_ADDONS_DIR" ]; then
+  echo "Error: Kodi addons dir does not exist at: $KODI_ADDONS_DIR" >&2
+  exit 1
+fi
+
+cd "$MACOSX_BINARY_ADDONS_TARGET_DIR"
+make
+rm -rf "$KODI_ADDONS_DIR/$ADDON_NAME"
+cp -rf "$XBMC_BUILD_ADDON_INSTALL_DIR" "$KODI_ADDONS_DIR"
```
Comment (AlwinEsch) [inline build-install-mac.sh]: A line break there

File: build-install-mac.sh [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+if [ ! -d "$BINARY_ADDONS_TARGET_DIR" ]; then
+  echo "Error: Could not find binary addons directory at: $BINARY_ADDONS_TARGET_DIR" >&2
+  exit 1
+fi
+
+XBMC_BUILD_ADDON_INSTALL_DIR=$(cd $1/addons/$ADDON_NAME 2> /dev/null && pwd -P)
+
+for DIR in "$BINARY_ADDONS_TARGET_DIR/"macosx*; do
+    if [ -d "${DIR}" ]; then
+	MACOSX_BINARY_ADDONS_TARGET_DIR="${DIR}"
```
Comment (AlwinEsch) [inline build-install-mac.sh]: If it is found can also a `break` added.

**Review by AlwinEsch — CHANGES_REQUESTED** [review]
Here a working way usable for all adding and not in conflict if some another is inside Kodi:
```sh
#!/bin/bash

set -e

if [ "$#" -ne 1 ] || ! [ -d "$1" ]; then
  echo "Usage: $0 <XBMC-SRC-DIR>" >&2
  exit 1
fi

if [[ "$OSTYPE" != "darwin"* ]]; then
  echo "Error: Script only for use on MacOSX" >&2
  exit 1
fi

BINARY_ADDONS_TARGET_DIR="$1/tools/depends/target/binary-addons"
BINARY_ADDON_SOURCE_DIR=`cd ..; pwd`
KODI_ADDONS_DIR="$HOME/Library/Application Support/Kodi/addons"
ADDON_NAME=`basename -s .git \`git config --get remote.origin.url\``

if [ ! -d "$KODI_ADDONS_DIR" ]; then
  echo "Error: Kodi addons dir does not exist at: $KODI_ADDONS_DIR" >&2
  exit 1
fi

cd "$1"
rm -f "$BINARY_ADDONS_TARGET_DIR"/.installed-macosx*
make -C tools/depends/target/binary-addons ADDONS=$ADDON_NAME ADDON_SRC_PREFIX="$BINARY_ADDON_SOURCE_DIR"

XBMC_BUILD_ADDON_INSTALL_DIR=$(cd $1/addons/$ADDON_NAME 2> /dev/null && pwd -P)

rm -rf "$KODI_ADDONS_DIR/$ADDON_NAME"
cp -rf "$XBMC_BUILD_ADDON_INSTALL_DIR" "$KODI_ADDONS_DIR"
```

File: README.md (hunk 5 above)
Comment (phunkyfish) [reply README.md:26]: For kodi yes, but for the pvr addons they all use xbmc.

File: build-install-mac.sh (hunk 3 above)
Comment (phunkyfish) [reply build-install-mac.sh:2]: Done

File: build-install-mac.sh (hunk 8 above)
Comment (phunkyfish) [reply build-install-mac.sh]: Done

File: build-install-mac.sh (hunk 1 above)
Comment (phunkyfish) [reply build-install-mac.sh]: Yup ;)

File: build-install-mac.sh (hunk 7 above)
Comment (phunkyfish) [reply build-install-mac.sh]: Done

File: build-install-mac.sh (hunk 4 above)
Comment (phunkyfish) [reply build-install-mac.sh]: Done

File: build-install-mac.sh (hunk 2 above)
Comment (phunkyfish) [reply build-install-mac.sh:43]: What about `make ADDONS=$ADDON_NAME ADDON_SRC_PREFIX=$HOME`

File: build-install-mac.sh [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+MACOSX_BINARY_ADDONS_TARGET_DIR=""
+KODI_ADDONS_DIR="$HOME/Library/Application Support/Kodi/addons"
+ADDON_NAME=`basename -s .git \`git config --get remote.origin.url\``
+
+if [ ! -d "$BINARY_ADDONS_TARGET_DIR" ]; then
+  echo "Error: Could not find binary addons directory at: $BINARY_ADDONS_TARGET_DIR" >&2
+  exit 1
+fi
+
+XBMC_BUILD_ADDON_INSTALL_DIR=$(cd $1/addons/$ADDON_NAME 2> /dev/null && pwd -P)
```
Comment (AlwinEsch) [inline build-install-mac.sh]: Also delete it then here ;)

File: build-install-mac.sh (hunk 2 above)
Comment (AlwinEsch) [reply build-install-mac.sh:43]: Is not required, seen on related CMakeCache.txt is it forced with ADDONS_TO_BUILD:UNINITIALIZED=pvr.iptvsimple

File: build-install-mac.sh (hunk 9 above)
Comment (phunkyfish) [reply build-install-mac.sh]: Ah, so cause I set it after make no need for it twice, gotcha.

Comment (phunkyfish) [reply build-install-mac.sh]: Ok, should be good now

**Review by AlwinEsch — APPROVED** [review]

**Comment by phunkyfish** [comment]
Will merge this shortly.

**Pull request merged**